Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -69,52 +66,495 @@ class UpstreamIntegrationTests: TestBase {
+        // Verify
+        assertExpectedEvents(ignoreUnexpectedEvents: true)
+    }
+    
+    /// Tests that a standard sendEvent receives the expected event handles
+    func testSendEvent_doesNotReceivesErrorEvent() {
+        // Setup
+        expectEdgeEventHandle(expectedHandleType: TestConstants.EventSource.LOCATION_HINT_RESULT, expectedCount: 1)
+        expectEdgeEventHandle(expectedHandleType: TestConstants.EventSource.STATE_STORE, expectedCount: 1)
 
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"], data: ["data": ["test": "data"]])
+
+        // Test
+        Edge.sendEvent(experienceEvent: experienceEvent)
+        
         // Verify
-        // MARK: Network response assertions
-        let networkRequest = NetworkRequest(urlString: "https://obumobile5.data.adobedc.net/ee/v1/interact", httpMethod: .post)!
-        let matchedResponsePost = networkService.getResponsesFor(networkRequest: networkRequest, timeout: 5)
-        XCTAssertEqual(200, matchedResponsePost.first?.responseCode)
-
-        // MARK: Response Event assertions
-        // Only validate for the location hint relevant to Edge Network extension
-        let expectedJSON = #"""
+        assertExpectedEvents(ignoreUnexpectedEvents: true)
+        
+        let errorEvents = getEdgeEventHandles(expectedHandleType: TestConstants.EventSource.ERROR_RESPONSE_CONTENT)
+        XCTAssertEqual(0, errorEvents.count)
+    }
+    
+    /// Tests that a standard sendEvent with no prior location hint value set receives the expected location hint event handle.
+    /// That is, checks for a string type location hint
+    func testSendEvent_with_NO_priorLocationHint_receivesExpectedLocationHintEventHandle() {
+        // Setup
+        // Clear any existing location hint
+        Edge.setLocationHint(nil)
+        
+        expectEdgeEventHandle(expectedHandleType: TestConstants.EventSource.LOCATION_HINT_RESULT, expectedCount: 1)
+
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"], data: ["data": ["test": "data"]])
+
+        // Test
+        Edge.sendEvent(experienceEvent: experienceEvent)
+        
+        // Verify
+        let expectedLocationHintJSON = #"""
         {
           "payload": [
             {
-              "ttlSeconds" : 1800,
+              "ttlSeconds" : 123,
               "scope" : "EdgeNetwork",
-              "hint" : "or2"
+              "hint" : "stringType"
+            }
+          ]
+        }
+        """#
+        
+        // See testSendEvent_receivesExpectedEventHandles for existence validation
+        let locationHintResult = getEdgeEventHandles(expectedHandleType: TestConstants.EventSource.LOCATION_HINT_RESULT).first!
+        
+        // Even though assertTypeMatch can accept a nil AnyCodable, expected uses ! because expectation is
+        // a test construct that should not be nil, and will be able to catch a nil actual
+        assertTypeMatch(expected: getAnyCodable(expectedLocationHintJSON)!,
```

Comment:
What if we mark the expected as required non-nil for the top level assert*Match APIs, this way we don't forget to provide a valid expected all the time. 
For nil expectations we can use XCTAssertNil(actual)

Reply:
looks good, let's also remove the comment above now since this is guided by the API now